feat(mcp): add configurable branding for MCP service

aminghadersohi opened the pull request:
### SUMMARY

Adds support for customizing MCP service branding by leveraging Superset's existing `APP_NAME` configuration, enabling white-label and enterprise deployments to use their own product names throughout the MCP service.

**Problem**: White-label and enterprise deployments of Superset need to present their own brand identity to AI assistants using the MCP service. Currently, all documentation and server names mention "Apache Superset" which may not match the deployed product branding.

**Solution**: Derive MCP branding from Superset's existing `APP_NAME` configuration instead of creating duplicate MCP-specific settings.

**How It Works**:
```python
# superset_config.py
APP_NAME = "ACME Analytics"

# Automatically results in:
# - MCP Server name: "ACME Analytics MCP Server"  
# - Instructions to AI: "You are connected to the ACME Analytics MCP service..."
```

**What Changes**:
- MCP server name shown to AI clients is derived as `f"{APP_NAME} MCP Server"`
- Product name throughout LLM instructions is derived from `APP_NAME`
- All references to "Apache Superset" become automatically branded

**Implementation**:
- `app.py`: Modified `init_fastmcp_server()` to read `APP_NAME` from Flask config
- `app.py`: Server name derived as `f"{APP_NAME} MCP Server"`
- `app.py`: Branding in instructions derived from `APP_NAME`
- `mcp_config.py`: Removed duplicate `MCP_SERVICE_NAME` and `MCP_SERVICE_BRANDING` constants

**Benefits**:
- ✅ **DRY**: Single source of truth for branding across all Superset services
- ✅ **Consistent**: Ensures UI, API, and MCP service use same branding
- ✅ **Simpler**: No additional MCP-specific configuration needed
- ✅ **Backwards Compatible**: Default `APP_NAME = "Superset"` maintains existing behavior

### BEFORE/AFTER

**Before** (default `APP_NAME = "Superset"`):
```
Server: "Superset MCP Server"
Instructions: "You are connected to the Apache Superset MCP service..."
```

**After** (with `APP_NAME = "ACME Analytics"`):
```
Server: "ACME Analytics MCP Server"
Instructions: "You are connected to the ACME Analytics MCP service..."
```

### TESTING INSTRUCTIONS

1. **Test default behavior** (no custom `APP_NAME`):
   ```bash
   superset mcp run --port 5008
   # Verify server name is "Superset MCP Server"
   # Verify LLM instructions mention "Apache Superset"
   ```

2. **Test custom branding** (ACME example):
   ```python
   # Add to superset_config.py
   APP_NAME = "ACME Analytics"
   ```
   ```bash
   superset mcp run --port 5008
   # Verify server name is "ACME Analytics MCP Server"
   # Verify LLM instructions mention "ACME Analytics" not "Apache Superset"
   ```

3. **Test OEM branding**:
   ```python
   # Add to superset_config.py
   APP_NAME = "DataViz Platform"
   ```
   ```bash
   superset mcp run --port 5008
   # Verify custom branding appears throughout
   ```

4. **Verify with AI client**:
   - Configure Claude Desktop or other MCP client
   - Connect to MCP service
   - Verify branded name appears in server listings
   - Ask a question and verify responses use branded product name

### ADDITIONAL INFORMATION

- [ ] Has associated issue: N/A
- [ ] Required feature flags: None
- [ ] Changes UI: No
- [ ] Includes DB Migration: No
- [x] Introduces new feature or API: Yes (derives branding from `APP_NAME`)
- [ ] Removes existing feature or API: No

---

**Note**: This approach addresses @rusackas's feedback to make branding configuration DRY by leveraging Superset's existing `APP_NAME` config instead of introducing duplicate MCP-specific settings.

aminghadersohi committed:
feat(mcp): add configurable branding for MCP service

Add support for customizing MCP service branding to enable white-label
deployments (e.g., Preset) to use their own product names.

**New Configuration Options** (superset_config.py):
- MCP_SERVICE_NAME: Server display name (default: "Superset MCP Server")
- MCP_SERVICE_BRANDING: Product name in LLM instructions (default: "Apache Superset")

**Changes**:
- mcp_config.py: Added branding configuration constants
- app.py: Made DEFAULT_INSTRUCTIONS parameterizable via get_default_instructions()
- app.py: Updated init_fastmcp_server() to read branding from Flask config
- BRANDING.md: Comprehensive documentation for branding configuration

**Use Cases**:
- Preset: Set MCP_SERVICE_BRANDING="Preset" for Preset-branded experience
- White-label: Custom product names for enterprise deployments
- Multi-tenant: Different branding per deployment

**Backwards Compatible**: Defaults to Apache Superset branding if not configured.

aminghadersohi committed:
refactor(mcp): use generic ACME examples instead of specific vendor names

Replace Preset-specific examples with generic ACME Corp examples to keep
branding documentation vendor-neutral and applicable to all deployments.

aminghadersohi committed:
fix(mcp): address PR review feedback - docstring format and use branding config

- Fix docstring formatting to put summary on same line as opening quotes
- Add branding parameter to create_mcp_app() to use MCP_SERVICE_BRANDING
- Include branding in MCP_FACTORY_CONFIG for proper configuration flow
- Addresses feedback from Bito and Korbit AI bots

rusackas commented:
Would it make sense for this config to extend parts of Superset's config? Then we can set certain properties on Superset, and the MCP service could leverage them (a bit more DRY)?

aminghadersohi committed:
refactor(mcp): derive branding from Superset's APP_NAME config

Address review feedback to make branding configuration DRY by deriving
from Superset's existing APP_NAME configuration instead of introducing
duplicate MCP-specific settings.

**Changes:**

- Removed `MCP_SERVICE_NAME` and `MCP_SERVICE_BRANDING` constants
- MCP server name now derived as "{APP_NAME} MCP Server"
- MCP branding in instructions derived directly from `APP_NAME`
- Updated `init_fastmcp_server()` to read `APP_NAME` from Flask config
- Updated `MCP_FACTORY_CONFIG` to use None (will derive at runtime)

**Example:**
```python
# superset_config.py
APP_NAME = "ACME Analytics"

# Results in:
# - Server name: "ACME Analytics MCP Server"
# - Instructions: "You are connected to the ACME Analytics MCP service..."
```

**Benefits:**
- Single source of truth for branding across all Superset services
- No duplicate configuration to maintain
- Ensures branding consistency (UI, API, MCP service)
- Simpler configuration for users

Addresses review feedback from @rusackas in PR #36033

aminghadersohi commented:
## ✅ Changes Pushed

Refactored to derive branding from `APP_NAME` as suggested. Changes pushed in commit 3aa3504.

**What changed:**
- ✅ Removed `MCP_SERVICE_NAME` and `MCP_SERVICE_BRANDING` constants
- ✅ MCP service now derives everything from `APP_NAME`
- ✅ Server name: `f"{APP_NAME} MCP Server"`
- ✅ Updated PR description to reflect simplified approach

**Example:**
```python
# superset_config.py
APP_NAME = "ACME Analytics"

# No MCP-specific config needed! Automatically uses:
# - Server name: "ACME Analytics MCP Server"
# - Branding: "ACME Analytics"
```

Much cleaner and more DRY! 🎉

dpgaspar commented on superset/mcp_service/mcp_config.py:
```diff
@@ -62,8 +62,9 @@
-    "instructions": None,  # Will use default from app.py
+    "name": None,  # Will derive from APP_NAME in app.py
+    "branding": None,  # Will derive from APP_NAME in app.py
+    "instructions": None,  # Will use default from app.py (parameterized with branding)
```
We could add a test with a branded `APP_NAME` asserting that instructions reflect this change

aminghadersohi committed:
test(mcp): add tests for APP_NAME branding configuration

Add comprehensive tests to verify that the MCP service properly derives
branding from Superset's APP_NAME configuration:

- Test get_default_instructions() with default and custom branding
- Test init_fastmcp_server() derives branding from APP_NAME
- Test server name is correctly formatted as "{APP_NAME} MCP Server"
- Test custom branding (e.g., "ACME Analytics") replaces default

This ensures white-label deployments can customize MCP service branding
without duplicate configuration.

aminghadersohi commented:
## ✅ Added Tests for APP_NAME Branding

As requested by @dpgaspar, I've added comprehensive tests to verify that the MCP service branding configuration works correctly with custom `APP_NAME` values.

### Tests Added (`tests/unit_tests/mcp_service/test_mcp_config.py`)

**1. Unit tests for `get_default_instructions()`:**
- ✅ Default branding produces "Apache Superset" 
- ✅ Custom branding (e.g., "ACME Analytics") replaces default
- ✅ Enterprise branding includes all tool documentation

**2. Integration tests for `init_fastmcp_server()`:**
- ✅ Default `APP_NAME="Superset"` produces "Superset" branding
- ✅ Custom `APP_NAME="ACME Analytics"` produces custom branding in instructions
- ✅ Server name is correctly derived as `"{APP_NAME} MCP Server"`

### Test Results
```
============================== test session starts ==============================
tests/unit_tests/mcp_service/test_mcp_config.py::test_get_default_instructions_with_default_branding PASSED
tests/unit_tests/mcp_service/test_mcp_config.py::test_get_default_instructions_with_custom_branding PASSED
tests/unit_tests/mcp_service/test_mcp_config.py::test_get_default_instructions_with_enterprise_branding PASSED
tests/unit_tests/mcp_service/test_mcp_config.py::test_init_fastmcp_server_with_default_app_name PASSED
tests/unit_tests/mcp_service/test_mcp_config.py::test_init_fastmcp_server_with_custom_app_name PASSED
tests/unit_tests/mcp_service/test_mcp_config.py::test_init_fastmcp_server_derives_server_name_from_app_name PASSED

============================== 6 passed in 1.00s ==============================
```

All tests pass and verify that the branding functionality works as expected! ✨

aminghadersohi committed:
fix(mcp-service): prevent Flask app creation during test imports

Fix test failures by mocking flask_singleton module before it's imported.
The tests were failing in CI because importing init_fastmcp_server caused
flask_singleton to create a real Flask app, which failed when auth_rate_limit
was None in the CI environment.

Changes:
- Use patch.dict to mock sys.modules["superset.mcp_service.flask_singleton"]
- Create mock Flask app before importing to avoid real app initialization
- Remove decorator-based patching that applied too late in the import chain

This prevents the AttributeError: 'NoneType' object has no attribute '__module__'
error that occurred when flask_limiter tried to access a None auth method.

aminghadersohi committed:
style: apply ruff formatting to test_mcp_config.py
